src/libsyntax/parse/parser.rs

```diff
@@ -5780,131 +5771,59 @@ impl<'a> Parser<'a> {
                     ty,
                     span,
                 });
-
-                seen_binding = true;
-                if first_type_or_binding_span.is_none() {
-                    first_type_or_binding_span = Some(span);
-                }
-                if first_binding_span.is_none() {
-                    first_binding_span = Some(span);
-                }
+                assoc_ty_bindings.push(span);
+            } else if self.check_const_arg() {
+                // Parse const argument.
+                let expr = if let token::OpenDelim(token::Brace) = self.token {
+                    self.parse_block_expr(None, self.span, BlockCheckMode::Default, ThinVec::new())?
+                } else if self.token.can_begin_literal_or_bool() {
+                    let lit = self.parse_lit()?;
```

Comment:
(So, I advised to use `parse_expr` here in `https://github.com/rust-lang/rust/pull/53645#discussion_r214497716`, but that was incorrect due to various issues around `>` not being treated as a closing angle bracket in expressions.)

Could you add tests for parsing `x<-1>` (pass) and `x<1 + 2>` (fail)?

Reply:
I've just noticed that `x<-1>` conflicts with emplacement syntax, so `x<-1>` currently fails. I'm not sure if it's reasonable to remove emplacement syntax now, but I'd imagine we'd want to do this before stabilising const generics.

Reply:
`x::<-1>` currently fails because `<-` is tokenised as a single token for emplacement syntax. For now, I simply have included this as a failing example, but presumably we'd want to remove emplacement syntax before const generics is stabilised.

Reply:
(I suggested IRL making `<-` get split into `<` and `-` the same way we do for e.g. `<<`)

Reply:
I was mostly interested in `x < - 1 >` passing than in dealing with the emplacement arrow, so could you add a test for that?